6.5.py

```diff
@@ -0,0 +1,14 @@
+def group_by(func, iterable) -> dict:
+    """
+    Function to create a dictionary. The key is the result of the function on the iterable,
+    and the value is list of the items that give this result.
+    :param func:
+    :param iterable:
+    :return:dictionary of keys and list of  iterable objects.
+    """
+    return {func(i): list(filter(lambda word: func(word) == func(i), iterable)) for i in iterable}
```

Comment:
I would recommend using default_dict - an elegant and efficient way of inserting unknown keys into a dictionary.
https://realpython.com/python-defaultdict/#:~:text=However%2C%20the%20defaultdict%20version%20is,instead%20of%20a%20standard%20dict%20.

Your code here is polynomial when it can be linear (you are actually doing the same action multiple times).